### DISCUSSION TITLE
Changing button note while running

### DISCUSSION THREAD

**Discussion by Mylab6** [tttapa/Control-Surface#414] [Q&A]
So for this example, 

// Instantiate a NoteButton object
NoteButton button = {
  7,                       // Push button on pin 5
  {note(C, 4), CHANNEL_1}, // Note C4 on MIDI channel 1
};
 

We can see the note is assigned when the button is created, anyway to update the button note later on, like during the loop. Also fantastic work !

**Comment by tttapa** [comment]
I believe this should answer your question: https://tttapa.github.io/Control-Surface-doc/new-input/Doxygen/df/df5/md_pages_FAQ.html#faq-change-address-runtime

If it doesn't or if you have any more questions, feel free to ask them here.

**Reply by Mylab6** [reply]
So in this example 

https://tttapa.github.io/Control-Surface-doc/new-input/Doxygen/d0/d74/Bank_8ino-example.html

I'm not clear on how I can change the bank without adding another input , am I missing something here ? 

This is what I want to accomplish
```

NoteButton button = {
    7,                      // Push button on pin 5
    {note(C, 4), CHANNEL_1}, // Note C4 on MIDI channel 1
};

void setup() {
button.note =  note(C,3) ; 
}

```

**Reply by tttapa** [reply]
You can use the `Bank::select()` method. For example:

```cpp
#include <Control_Surface.h>
using namespace MIDI_Notes;
USBDebugMIDI_Interface midi;
OutputBank bank {1, note(C, 4)}; // Bank determines the address offset, resolution = 1, initial address = note C4
Bankable::NoteButton notebutton {bank, 7, {0, CHANNEL_1}}; // Set base address to 0, bank has full control of the address
Button button {6}; // Button to test if changing the address in the loop() works

void setup() {
  Control_Surface.begin();
  button.begin();
}

void loop() {
  Control_Surface.loop();
  if (button.update() == Button::Falling) {
    bank.select(note(C, 3));
  } else if (button.getState() == Button::Rising) {
    bank.select(note(C, 4));
  }
}
```
It's probably misusing the bank system a little bit, but it should work. If you need finer control over the addresses, it might be worth looking at the [Custom-MIDI-Output-Element](https://tttapa.github.io/Control-Surface-doc/Doxygen/d4/d3b/Custom-MIDI-Output-Element_8ino-example.html) example.
